stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -12,28 +12,267 @@
+/// Since those methods are not currently expressible as protocol requirements Swift, and therefore will not appear in
+/// the protocol's documentation as explicit requirements, due to their advanced use of generics, we present their
+/// signatures that a conforming type one has to implement here:
+///
+/// > Note: Although the the `remoteCall` methods are not expressed as protocol requirements in source,
+/// > the compiler will provide the same errors as-if they were declared explicitly in this protocol.
+///
+/// ```swift
+///  /// Invoked by the Swift runtime when making a remote call.
+///  ///
+///  /// The `arguments` are the arguments container that was previously created
+///  /// by `makeInvocationEncoder` and has been populated with all arguments.
+///  ///
+///  /// This method should perform the actual remote function call, and await for its response.
+///  ///
+///  /// ## Errors
+///  /// This method is allowed to throw because of underlying transport or serialization errors,
+///  /// as well as by re-throwing the error received from the remote callee (if able to).
+/// func remoteCall<Act, Err, Res>(
+///     on actor: Act,
+///     target: RemoteCallTarget,
+///     invocation: inout InvocationEncoder,
+///     throwing: Err.Type,
+///     returning: Res.Type
+/// ) async throws -> Res
+///     where Act: DistributedActor,
+///           Act.ID == ActorID,
+///           Err: Error,
+///           Res: SerializationRequirement
+/// ```
+///
+/// ```swift
+///  /// Invoked by the Swift runtime when making a `Void`-returning remote call.
+///  ///
+///  /// The `arguments` are the arguments container that was previously created
+///  /// by `makeInvocationEncoder` and has been populated with all arguments.
+///  ///
+///  /// This method should perform the actual remote function call, and await for its response.
+///  ///
+///  /// ## Errors
+///  /// This method is allowed to throw because of underlying transport or serialization errors,
+///  /// as well as by re-throwing the error received from the remote callee (if able to).
+/// func remoteCallVoid<Act, Err>(
+///     on actor: Act,
+///     target: RemoteCallTarget,
+///     invocation: inout InvocationEncoder,
+///     throwing: Err.Type
+/// ) async throws -> Res
+///     where Act: DistributedActor,
+///           Act.ID == ActorID,
+///           Err: Error
+/// ```
+///
+/// Implementations of remote calls generally will serialize the `actor.id`, `target` and `invocation`
+/// into some form of wire envelope, and send it over the network (or process boundary) using some
+/// transport mechanism of their choice. As they do so, they need to suspend the remoteCall function,
+/// and resume it once a reply to the call arrives; Unless the transport layer is also async/await aware,
+/// this will often require making use of a ``CheckedContinuation``.
```

Comment:
```suggestion
/// Implementations of remote calls generally will serialize `actor.id`, `target` and `invocation`
/// into some form of wire envelope, and send it over the network (or process boundary) using some
/// transport mechanism of their choice. As they do so, they need to suspend the `remoteCall` function,
/// and resume it once a reply to the call arrives. Unless the transport layer is also async/await aware,
/// this will often require making use of a ``CheckedContinuation``.
```